### ISSUE TITLE
no-extra-parens false positives with new and member expressions

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12740]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2015
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogXCJlcnJvclwiICovXG5cbm5ldyAoYSgpKS5iLmM7XG5cbm5ldyAoYSgpLmIpLmM7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

```js
/* eslint no-extra-parens: "error" */

new (a()).b.c;

new (a().b).c;
```

```bash
eslint index.js --fix
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

2 errors, and auto-fixed to:

```js
/* eslint no-extra-parens: "error" */

new a().b.c;

new a().b.c;
```

The original expressions are both equivalent to `new (a().b.c)`

The expressions produced in the fix are equivalent to `(new a()).b.c`, which is a different thing.

**Are you willing to submit a pull request to fix this bug?**

Yes, but please verify is this a bug.

It seems that `MemberExpression(node)` should do a better check for both `object` and `parent`.

(also, this is not a regression from #12302, it's unrelated).
